auth_mellon_handler.c

```diff
@@ -807,6 +807,54 @@ static int am_handle_logout_request(request_rec *r,
     return rc;
 }
 
+/* This function handles an invalidate request.
+ *
+ * Parameters:
+ *  request_rec *r       The logout request.
+ *
+ * Returns:
+ *  OK on success, or an error if any of the steps fail.
+ */
+static int am_handle_invalidate_request(request_rec *r)
+{
+    int rc;
+    char *return_to;
+    am_cache_entry_t *session = am_get_request_session(r);
+
+    am_diag_printf(r, "enter function %s\n", __func__);
+    am_diag_log_cache_entry(r, 0, session, "%s\n", __func__);
+
+    if(session != NULL) {
+        am_delete_request_session(r, session);
+    }
+
+    return_to = am_extract_query_parameter(r->pool, r->args, "ReturnTo");
+
+    if(return_to == NULL) {
+        AM_LOG_RERROR(APLOG_MARK, APLOG_ERR, 0, r,
+                      "No ReturnTo parameter provided for invalidate handler.");
+        return HTTP_BAD_REQUEST;
```

Comment:
if the request is malformed, should you still delete the session anyway ?

Reply:
if yes then I would not consider the req malformed and simply redirect to the referer or missing that the main mod_mellon page ?